ffs/minerselector/reptop/reptop.go

```diff
@@ -29,24 +29,36 @@ func New(rm *reputation.Module, ai *askRunner.Runner) *RepTop {
 // GetMiners returns n miners using the configured Reputation Module and
 // Ask Index.
 func (rt *RepTop) GetMiners(n int, f ffs.MinerSelectorFilter) ([]ffs.MinerProposal, error) {
-	ms, err := rt.rm.QueryMiners(n, f.ExcludedMiners, f.CountryCodes, f.TrustedMiners)
+	if n < 1 {
+		return nil, fmt.Errorf("the number of miners should be greater than zero")
+	}
+	ms, err := rt.rm.QueryMiners(f.ExcludedMiners, f.CountryCodes, f.TrustedMiners)
 	if err != nil {
-		return nil, fmt.Errorf("getting top %d miners from reputation module: %s", n, err)
+		return nil, fmt.Errorf("getting miners from reputation module: %s", err)
 	}
 	if len(ms) < n {
 		return nil, fmt.Errorf("not enough miners that satisfy the constraints")
 	}
 	aidx := rt.ai.Get()
-	res := make([]ffs.MinerProposal, 0, len(ms))
+	res := make([]ffs.MinerProposal, 0, n)
 	for _, m := range ms {
 		sa, ok := aidx.Storage[m.Addr]
 		if !ok {
 			continue
 		}
+		if f.MaxPrice > 0 && sa.Price > f.MaxPrice {
+			continue
+		}
```

Comment:
If max price isn't the default (0, which means no max price), compare the ask price to the max price and ignore the ask if the price is too high.